pybamm/experiments/experiment.py

```diff
@@ -45,6 +45,13 @@ class Experiment:
 
     def __init__(self, operating_conditions, parameters=None, period="1 minute"):
         self.period = self.convert_time_to_seconds(period.split())
+        if all([isinstance(cycle, tuple) for cycle in operating_conditions]):
+            self.cycle_lengths = [len(cycle) for cycle in operating_conditions]
+            operating_conditions = [
+                cond for cycle in operating_conditions for cond in cycle
+            ]
+        elif all([isinstance(cond, str) for cond in operating_conditions]):
+            self.cycle_lengths = [len(operating_conditions)]
```

Comment:
What about the case where there is a mix of tuples and strings? This should either raise an error, or strings should be made into 1-tuples (i.e. cycles of length 1). Second option slightly better IMO.

Reply:
I thought it made sense to keep all the type-checking together, so I moved the check for strings out of experiment.read_operating_conditions and do it all in __init__. Hopefully my solution is not too opaque (it got a bit complicated checking both cycles and the operating conditions therein).